exporter/containerimage/writer.go

```diff
@@ -369,8 +388,8 @@ func (ic *ImageWriter) exportLayers(ctx context.Context, refCfg cacheconfig.RefC
 // the new blob.
 //
 // If no conversion is needed, this returns nil without error.
-func rewriteImageLayerWithEpoch(ctx context.Context, cs content.Store, desc ocispecs.Descriptor, comp compression.Config, epoch *time.Time) (*ocispecs.Descriptor, error) {
-	converterFn, err := converter.NewWithRewriteTimestamp(ctx, cs, desc, comp, epoch)
+func rewriteImageLayerWithEpoch(ctx context.Context, cs content.Store, desc ocispecs.Descriptor, comp compression.Config, epoch *time.Time, immDiffIDs map[digest.Digest]struct{}) (*ocispecs.Descriptor, error) {
```

Comment:
If this function is for a single blob (without chain) then all base chain checks should happen before it is called.

Reply:
Fixed